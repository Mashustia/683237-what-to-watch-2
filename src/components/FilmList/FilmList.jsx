import React, {Component} from 'react';
import PropTypes from 'prop-types';

import FilmCard from '../FilmCard/FilmCard';

class FilmList extends Component {
  constructor(props) {
    super(props);

    this.state = {
      inFocus: ``
    };

    this.onFilmFocus = this.onFilmFocus.bind(this);
  }

  onFilmFocus(id) {
    this.setState({
      inFocus: id
    });
  }

  render() {
    const {films} = this.props;

    return (
      <div className='catalog__movies-list'>
        {films.map((film) => <FilmCard
          film={film.name}
          key={`${film.id}`}
          id={film.id}
          onClick={() => {}}
          onFilmFocus={this.onFilmFocus}
        />)}
      </div>
    );
  }
}

FilmList.propTypes = {
  films: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
        name: PropTypes.string.isRequired,
        genre: PropTypes.string.isRequired,
        description: PropTypes.string,
        director: PropTypes.string,
        starring: PropTypes.string,
        rating: PropTypes.string,
        runTime: PropTypes.string,
        released: PropTypes.string,
        reviews: PropTypes.array,
        cover: PropTypes.string,
      })
  )
};

export default FilmList;
