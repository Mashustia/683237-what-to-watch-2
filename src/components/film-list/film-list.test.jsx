import React from 'react';
import renderer from 'react-test-renderer';

import FilmList from './film-list';

it(`FilmList component renders correctly`, () => {
  const component = renderer
    .create(<FilmList films={[{
      id: `1`,
      name: `F`,
      genre: ``,
      description: ``,
      director: ``,
      starring: ``,
      rating: ``,
      runTime: ``,
      released: ``,
      reviews: [],
      cover: ``,
      preview: ``
    }]} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});
