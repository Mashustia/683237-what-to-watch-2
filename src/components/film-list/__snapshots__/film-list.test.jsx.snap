// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmList component renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseOut={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="F"
        height="175"
        src="img/f.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        F
      </a>
    </h3>
    <div
      className="film-hide"
    >
      <video
        className="preview"
        controls={true}
        src=""
      />
    </div>
  </article>
</div>
`;
